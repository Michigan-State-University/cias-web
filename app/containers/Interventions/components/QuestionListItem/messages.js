import { defineMessages } from 'react-intl';

export const scope = 'app.containers.QuestionListItem';

export default defineMessages({
  delete: {
    id: `${scope}.delete`,
    defaultMessage: 'Delete',
  },
  copy: {
    id: `${scope}.copy`,
    defaultMessage: 'Copy',
  },
});
