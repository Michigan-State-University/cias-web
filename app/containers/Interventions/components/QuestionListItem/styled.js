import styled from 'styled-components';
import React from 'react';
import PropTypes from 'prop-types';
import { colors, themeColors } from 'theme';
import Circle from 'components/Circle';
import { fontSizes } from 'theme/fonts';
import HoverableBox from 'components/Box/HoverableBox';
import H3 from 'components/H3';

const NumberCircle = props => (
  <Circle
    {...props}
    size="23px"
    bg={themeColors.primary}
    fontSize={fontSizes.small}
    {...(props.isSelected
      ? { textOpacity: 1, color: colors.white }
      : { textOpacity: 0.3, color: colors.jungleGreen })}
  />
);

NumberCircle.propTypes = {
  isSelected: PropTypes.bool,
};

const ToggleableBox = styled(HoverableBox)`
  ${props => (props.isSelected ? { backgroundColor: colors.zirkon } : {})};
`;

ToggleableBox.propTypes = {
  isSelected: PropTypes.bool,
};

const ClampedTitle = styled(H3)`
  -webkit-line-clamp: 2;
  overflow: hidden;
  -webkit-box-orient: vertical;
  -moz-box-orient: vertical;
  -ms-box-orient: vertical;
  -o-box-orient: vertical;
  display: -webkit-box;
`;

export { NumberCircle, ToggleableBox, ClampedTitle };
