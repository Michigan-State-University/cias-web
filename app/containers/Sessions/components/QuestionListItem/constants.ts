import { QuestionTypes } from 'models/Question';

export const NON_MANAGEABLE_SCREENS = [
  QuestionTypes.FINISH,
  QuestionTypes.TLFB_CONFIG,
  QuestionTypes.TLFB_QUESTION,
  QuestionTypes.TLFB_EVENTS,
];

export const NON_DUPLICABLE_SCREENS = [
  QuestionTypes.NAME,
  QuestionTypes.PARTICIPANT_REPORT,
  QuestionTypes.PHONE,
  QuestionTypes.HENRY_FORD_INITIAL,
];

// screens whose variable is not editable
export const VARIABLE_NON_EDITABLE_SCREENS = [QuestionTypes.NAME];

export const ONLY_NARRATOR_TAB_SCREENS = [
  QuestionTypes.TLFB_QUESTION,
  QuestionTypes.TLFB_EVENTS,
];
