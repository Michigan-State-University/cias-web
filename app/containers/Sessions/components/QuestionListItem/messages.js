import { defineMessages } from 'react-intl';

export const scope = 'app.containers.QuestionListItem';

export default defineMessages({
  delete: {
    id: `${scope}.delete`,
    defaultMessage: 'Delete',
  },
  copy: {
    id: `${scope}.copy`,
    defaultMessage: 'Copy',
  },
  duplicate: {
    id: `${scope}.duplicate`,
    defaultMessage: 'Duplicate',
  },
  pasteQuestion: {
    id: `${scope}.pasteQuestion`,
    defaultMessage: 'Paste question in this group',
  },
});
