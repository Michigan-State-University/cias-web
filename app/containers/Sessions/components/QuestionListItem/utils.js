const getIndex = (selectedQuestionIndex, questionsLength) => {
  // when user tries to delete first question
  if (selectedQuestionIndex === 0 && questionsLength > 1) {
    return 0;
  }
  // when user tries to delete last question
  if (selectedQuestionIndex === questionsLength - 1 && questionsLength > 1) {
    return selectedQuestionIndex - 1;
  }
  // when user tries to delete second to last question
  if (selectedQuestionIndex === questionsLength - 2) {
    return selectedQuestionIndex;
  }
  // when user tries to delete one and only question
  if (questionsLength === 1) {
    return 0;
  }
  // when user tries to delete question from the middle of the list
  return selectedQuestionIndex + 1;
};

export default getIndex;
