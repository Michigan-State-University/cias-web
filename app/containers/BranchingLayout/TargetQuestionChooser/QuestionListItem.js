import React from 'react';
import PropTypes from 'prop-types';

import Img from 'components/Img';
import Box from 'components/Box';
import EllipsisText from 'components/Text/EllipsisText';
import Row from 'components/Row';

import webpageSelected from 'assets/svg/webpage-mouseover-selected.svg';
import webpage from 'assets/svg/webpage-mouseover.svg';
import { htmlToPlainText } from 'utils/htmlToPlainText';
import { colors } from 'theme';

const QuestionListItem = ({
  onClick,
  target,
  index,
  selectedQuestionId,
  question: { id, subtitle, type },
}) => {
  const canSelectQuestion = questionId => selectedQuestionId !== questionId;

  return (
    <Row
      data-testid={`${selectedQuestionId}-select-target-question-el`}
      key={`${selectedQuestionId}-select-target-question-${index}`}
      px={10}
      mt={20}
      onClick={() => canSelectQuestion(id) && onClick({ type, id })}
      clickable={canSelectQuestion(id)}
      width="100%"
    >
      <Img src={target.id === id ? webpageSelected : webpage} mr={10} />
      <Box maxWidth={230}>
        <EllipsisText
          text={htmlToPlainText(subtitle)}
          color={!canSelectQuestion(id) ? colors.grey : ''}
          fontWeight={target.id === id ? 'bold' : ''}
        />
      </Box>
    </Row>
  );
};

QuestionListItem.propTypes = {
  onClick: PropTypes.func,
  target: PropTypes.object,
  index: PropTypes.number,
  selectedQuestionId: PropTypes.string,
  question: PropTypes.shape({
    id: PropTypes.string,
    subtitle: PropTypes.string,
    type: PropTypes.string,
  }),
};

export default QuestionListItem;
