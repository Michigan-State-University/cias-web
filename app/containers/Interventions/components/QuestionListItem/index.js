import React from 'react';
import PropTypes from 'prop-types';
import cloneDeep from 'lodash/cloneDeep';
import set from 'lodash/set';
import uniqueId from 'lodash/uniqueId';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { injectIntl, FormattedMessage } from 'react-intl';
import unescape from 'lodash/unescape';

import Badge from 'components/Badge';
import Column from 'components/Column';
import Comment from 'components/Text/Comment';
import Dropdown from 'components/Dropdown';
import Row from 'components/Row';
import globalMessages from 'global/i18n/globalMessages';
import { colors } from 'theme';
import { getNarratorPositionWhenQuestionIsChanged } from 'utils/getNarratorPosition';
import { hasObjectProperty } from 'utils/hasObjectProperty';
import { htmlToPlainText } from 'utils/htmlToPlainText';
import {
  selectQuestion,
  deleteQuestionRequest,
  copyQuestionRequest,
} from 'global/reducers/questions';
import {
  setQuestionSettings,
  makeSelectQuestionSettingsVisibility,
  setAnimationStopPosition,
  setCharacterDraggable,
  changeCurrentNarratorBlock,
} from 'global/reducers/localState';

import StyledCircle from 'components/Circle/StyledCircle';
import { QuestionTypes } from 'models/Intervention/QuestionTypes';
import Box from 'components/Box';
import Radio from 'components/Radio';
import { ToggleableBox, ClampedTitle } from './styled';
import messages from './messages';
import getIndex from './utils';

const QuestionListItem = ({
  question,
  index,
  onSelect,
  selectedQuestionIndex,
  questions,
  toggleSettings,
  removeQuestion,
  intl: { formatMessage },
  copyQuestion,
  interventionId,
  changeNarratorBlockIndex,
  setDraggable,
  setCharacterPosition,
  checked,
  selectSlide,
  manage,
  disabled,
}) => {
  const { type, subtitle, id, body } = question;
  const isSelected = selectedQuestionIndex === id;

  const handleSelectClick = newIndex => {
    setDraggable(false);
    if (selectedQuestionIndex !== id) {
      onSelect(id);
      const newPosition = getNarratorPositionWhenQuestionIsChanged(
        questions,
        newIndex,
      );
      setCharacterPosition(newPosition.x, newPosition.y);
    }
  };

  const handleDelete = () => {
    const newIndex = getIndex(selectedQuestionIndex, questions.length);
    handleSelectClick(newIndex);
    removeQuestion({ questionId: id, interventionId });
  };

  const handleCopy = () => {
    const copied = cloneDeep(question);
    set(copied, 'id', uniqueId());
    copyQuestion({ copied, questionId: id });
  };

  const options = [
    {
      id: 'delete',
      label: <FormattedMessage {...messages.delete} />,
      action: handleDelete,
      color: colors.flamingo,
    },
    {
      id: 'copy',
      label: <FormattedMessage {...messages.copy} />,
      action: handleCopy,
      color: colors.black,
    },
  ];

  const onChangeItem = () => {
    handleSelectClick(index);
    changeNarratorBlockIndex(-1);
    toggleSettings({ index, questionIndex: selectedQuestionIndex });
  };

  return (
    <ToggleableBox
      padding={15}
      mb={15}
      width="100%"
      onClick={onChangeItem}
      isSelected={isSelected}
      bg="#F8FBFF"
      border={`1px solid ${checked ? '#C866EA' : '#EFEFEF'}`}
    >
      <Row justify="between">
        {manage && (
          <Column xs={1}>
            <Radio
              onClick={e => {
                selectSlide(id);
                e.stopPropagation();
              }}
              checked={checked}
            />
          </Column>
        )}
        <Column xs={10}>
          <Row>
            <ClampedTitle mb={6}>
              {unescape(htmlToPlainText(subtitle))}
            </ClampedTitle>
          </Row>
          <Row>
            <Box display="flex" align="center">
              <StyledCircle
                background={
                  QuestionTypes.find(({ id: typeId }) => typeId === type).color
                }
                size="10px"
                mr="5px"
              />
              <Comment fontWeight="bold">
                {formatMessage(globalMessages.questionTypes[type])}
              </Comment>
            </Box>
          </Row>
          {body && hasObjectProperty(body, 'variable') && (
            <Row mt={5}>
              <Badge color={colors.jungleGreen} bgWithOpacity>
                {(body.variable.name && body.variable.name.trim()) ||
                  formatMessage(globalMessages.variables.emptyVariable)}
              </Badge>
            </Row>
          )}
        </Column>
        {!manage && !disabled && (
          <Column xs={1}>
            <Dropdown options={options} />
          </Column>
        )}
      </Row>
    </ToggleableBox>
  );
};

QuestionListItem.propTypes = {
  question: PropTypes.object.isRequired,
  index: PropTypes.number.isRequired,
  onSelect: PropTypes.func.isRequired,
  selectedQuestionIndex: PropTypes.string,
  questions: PropTypes.array,
  settingsVisibility: PropTypes.bool,
  toggleSettings: PropTypes.func,
  copyQuestion: PropTypes.func,
  intl: PropTypes.object,
  removeQuestion: PropTypes.func,
  interventionId: PropTypes.string,
  changeNarratorBlockIndex: PropTypes.func,
  setDraggable: PropTypes.func,
  setCharacterPosition: PropTypes.func,
  checked: PropTypes.bool,
  manage: PropTypes.bool,
  selectSlide: PropTypes.func,
  disabled: PropTypes.bool,
};

const mapStateToProps = createStructuredSelector({
  settingsVisibility: makeSelectQuestionSettingsVisibility(),
});

const mapDispatchToProps = {
  onSelect: selectQuestion,
  toggleSettings: setQuestionSettings,
  removeQuestion: deleteQuestionRequest,
  copyQuestion: copyQuestionRequest,
  changeNarratorBlockIndex: changeCurrentNarratorBlock,
  setDraggable: setCharacterDraggable,
  setCharacterPosition: setAnimationStopPosition,
};

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default injectIntl(compose(withConnect)(QuestionListItem));
