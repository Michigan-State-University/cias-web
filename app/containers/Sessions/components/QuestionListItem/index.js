import React, { useEffect, useMemo, useRef, useState } from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { FormattedMessage, injectIntl } from 'react-intl';
import { Draggable } from 'react-beautiful-dnd';
import cloneDeep from 'lodash/cloneDeep';
import set from 'lodash/set';
import uniqueId from 'lodash/uniqueId';

import Column from 'components/Column';
import Comment from 'components/Text/Comment';
import Dropdown from 'components/Dropdown';
import Row from 'components/Row';
import CopyModal from 'components/CopyModal';
import globalMessages from 'global/i18n/globalMessages';
import { colors, themeColors } from 'theme';
import { getNarratorPositionWhenQuestionIsChanged } from 'utils/getNarratorPosition';
import { hasObjectProperty } from 'utils/hasObjectProperty';
import { htmlToPlainText } from 'utils/htmlToPlainText';
import {
  copyQuestionRequest,
  deleteQuestionRequest,
  makeSelectQuestions,
  selectQuestion,
  copyExternallyQuestionRequest,
  makeSelectLastCreatedQuestionId,
} from 'global/reducers/questions';
import {
  changeCurrentNarratorBlock,
  makeSelectQuestionSettingsVisibility,
  setAnimationStopPosition,
  setCharacterDraggable,
  setQuestionSettings,
} from 'global/reducers/localState';

import StyledCircle from 'components/Circle/StyledCircle';
import {
  finishQuestion,
  QuestionTypes,
  nameQuestion,
  participantReport,
  phoneQuestion,
} from 'models/Session/QuestionTypes';
import Box from 'components/Box';
import Checkbox from 'components/Checkbox';
import ConfirmationBox from 'components/ConfirmationBox';
import Text from 'components/Text';
import scrollByRef from 'utils/scrollByRef';
import VariableInput from '../QuestionDetails/VariableInput';
import { ClampedTitle, ToggleableBox } from './styled';
import messages from './messages';
import getIndex from './utils';

const QuestionListItem = ({
  question,
  index,
  onSelect,
  selectedQuestionIndex,
  questions,
  toggleSettings,
  removeQuestion,
  intl: { formatMessage },
  copyQuestion,
  copyExternallyQuestion,
  changeNarratorBlockIndex,
  setDraggable,
  setCharacterPosition,
  checked,
  selectSlide,
  manage,
  interventionStatus,
  disabled,
  noDnd,
  groupIds,
  allQuestions,
  sessionId,
  lastCreatedQuestionId,
}) => {
  const questionRef = useRef(null);
  const [copyOpen, setCopyOpen] = useState(false);
  const [deleteOpen, setDeleteOpen] = useState(false);
  const { type, subtitle, id, body, question_group_id: groupId } = question;
  const isSelected = selectedQuestionIndex === id;

  const isFinishScreen = useMemo(() => type === finishQuestion.id, [type]);
  const isNameScreen = useMemo(() => type === nameQuestion.id, [type]);

  const canDuplicate = useMemo(
    () =>
      type !== nameQuestion.id &&
      type !== participantReport.id &&
      type !== phoneQuestion.id,
    [type],
  );

  useEffect(() => {
    if (selectedQuestionIndex === id) beforeRender();
  }, [selectedQuestionIndex]);

  useEffect(() => {
    if (
      lastCreatedQuestionId === id &&
      selectedQuestionIndex === id &&
      questionRef.current
    )
      scrollByRef(questionRef, true);
  }, [lastCreatedQuestionId, selectedQuestionIndex]);

  const handleSelectClick = () => {
    document.activeElement.blur();
    setDraggable(false);
    if (selectedQuestionIndex !== id) {
      onSelect(id);
      const newPosition = getNarratorPositionWhenQuestionIsChanged(
        allQuestions,
        id,
        groupIds,
      );
      setCharacterPosition(newPosition.x, newPosition.y);
    }
  };

  const handleDelete = () => {
    const newIndex = getIndex(selectedQuestionIndex, questions.length);
    handleSelectClick(newIndex);
    removeQuestion({ questionId: id, groupId, groupIds, sessionId });
  };

  const closeCopyModal = () => setCopyOpen(false);

  const handleCopyModal = () => {
    setCopyOpen(true);
  };

  const handleCopy = (target) => {
    const copied = cloneDeep(question);
    set(copied, 'id', uniqueId());
    copyQuestion({ copied, questionId: question.id, target });
  };

  const handleExternallyCopy = (target) => {
    const copied = cloneDeep(question);
    set(copied, 'id', uniqueId());
    copyExternallyQuestion(target.sessionId, target.id, copied, [question.id]);
  };

  const options = [
    {
      id: 'copy',
      label: <FormattedMessage {...messages.copy} />,
      action: handleCopyModal,
      color: colors.black,
    },
    {
      id: 'duplicate',
      label: <FormattedMessage {...messages.duplicate} />,
      action: handleCopy,
      color: colors.black,
      disabled: disabled || !canDuplicate,
    },
    {
      id: 'delete',
      label: <FormattedMessage {...messages.delete} />,
      action: () => setDeleteOpen(true),
      color: themeColors.warning,
      disabled,
    },
  ];

  const onChangeItem = () => {
    handleSelectClick(index);
  };

  const beforeRender = () => {
    setDraggable(false);
    changeNarratorBlockIndex(-1);
    toggleSettings({ index, questionIndex: selectedQuestionIndex });
  };

  const renderQuestion = () => (
    <>
      <CopyModal
        visible={copyOpen}
        onClose={closeCopyModal}
        copyAction={handleExternallyCopy}
        disableInterventionCopy
        disableSessionCopy
        pasteText={formatMessage(messages.pasteQuestion)}
      />
      <ConfirmationBox
        visible={deleteOpen}
        onClose={() => setDeleteOpen(false)}
        description={formatMessage(messages.deleteModalTitle)}
        content={
          <Column align="center">
            <Text color={themeColors.warning}>
              {formatMessage(messages.deleteModalContent)}
            </Text>
          </Column>
        }
        confirmAction={handleDelete}
      />
      <ToggleableBox
        padding={15}
        mb={15}
        width="100%"
        onClick={onChangeItem}
        isSelected={isSelected}
        bg={colors.zirkon}
        border={`1px solid ${
          checked ? colors.electricPurple : colors.smokeWhite
        }`}
      >
        <Row justify="between" ref={questionRef}>
          {manage && !isFinishScreen && (
            <Column xs={1}>
              <Checkbox
                onClick={(e) => {
                  selectSlide(id);
                  e.stopPropagation();
                  e.preventDefault();
                }}
                checked={checked}
              />
            </Column>
          )}
          <Column xs={10}>
            <Row>
              <ClampedTitle mb={6}>{htmlToPlainText(subtitle)}</ClampedTitle>
            </Row>
            <Row>
              <Box display="flex" align="center">
                <StyledCircle
                  background={
                    QuestionTypes.find(({ id: typeId }) => typeId === type)
                      .color
                  }
                  size="10px"
                  mr="5px"
                />
                <Comment fontWeight="bold">
                  {formatMessage(globalMessages.questionTypes[type])}
                </Comment>
              </Box>
            </Row>
            {body && hasObjectProperty(body, 'variable') && (
              <Row mt={10}>
                <VariableInput
                  questionId={id}
                  variable={body.variable}
                  interventionStatus={interventionStatus}
                  disabled={isNameScreen}
                />
              </Row>
            )}
          </Column>
          {!manage && !isFinishScreen && (
            <Column xs={1}>
              <Dropdown options={options} />
            </Column>
          )}
        </Row>
      </ToggleableBox>
    </>
  );

  const renderQuestionWithDnd = () => (
    <Draggable
      key={`group-${groupId}-item-${id}`}
      draggableId={id}
      index={index}
      isDragDisabled={disabled}
    >
      {(provided) => (
        <Box
          width="100%"
          key={id}
          ref={provided.innerRef}
          {...provided.draggableProps}
          {...provided.dragHandleProps}
        >
          {renderQuestion()}
        </Box>
      )}
    </Draggable>
  );

  return noDnd ? renderQuestion() : renderQuestionWithDnd();
};

QuestionListItem.propTypes = {
  question: PropTypes.object.isRequired,
  index: PropTypes.number.isRequired,
  onSelect: PropTypes.func.isRequired,
  selectedQuestionIndex: PropTypes.string,
  questions: PropTypes.array,
  settingsVisibility: PropTypes.bool,
  toggleSettings: PropTypes.func,
  copyQuestion: PropTypes.func,
  copyExternallyQuestion: PropTypes.func,
  intl: PropTypes.object,
  removeQuestion: PropTypes.func,
  sessionId: PropTypes.string,
  changeNarratorBlockIndex: PropTypes.func,
  setDraggable: PropTypes.func,
  setCharacterPosition: PropTypes.func,
  checked: PropTypes.bool,
  manage: PropTypes.bool,
  selectSlide: PropTypes.func,
  disabled: PropTypes.bool,
  interventionStatus: PropTypes.string,
  noDnd: PropTypes.bool,
  groupIds: PropTypes.array,
  allQuestions: PropTypes.array,
  lastCreatedQuestionId: PropTypes.string,
};

const mapStateToProps = createStructuredSelector({
  settingsVisibility: makeSelectQuestionSettingsVisibility(),
  allQuestions: makeSelectQuestions(),
  lastCreatedQuestionId: makeSelectLastCreatedQuestionId(),
});

const mapDispatchToProps = {
  onSelect: selectQuestion,
  toggleSettings: setQuestionSettings,
  removeQuestion: deleteQuestionRequest,
  copyQuestion: copyQuestionRequest,
  copyExternallyQuestion: copyExternallyQuestionRequest,
  changeNarratorBlockIndex: changeCurrentNarratorBlock,
  setDraggable: setCharacterDraggable,
  setCharacterPosition: setAnimationStopPosition,
};

const withConnect = connect(mapStateToProps, mapDispatchToProps);

export default injectIntl(compose(withConnect)(QuestionListItem));
