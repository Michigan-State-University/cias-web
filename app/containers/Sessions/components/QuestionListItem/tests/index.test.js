import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { IntlProvider } from 'react-intl';
import { DEFAULT_LOCALE } from 'i18n';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import configureStore from 'configureStore';

import { singleQuestion } from 'models/Session/QuestionTypes';
import { withDroppable } from 'utils/testUtils/dndUtils';

import QuestionListItem from '../index';

describe('<QuestionListItem />', () => {
  let store;
  let modalContainer;

  beforeAll(() => {
    store = configureStore({}, browserHistory);

    ReactDOM.createPortal = jest.fn((element) => element);
    modalContainer = document.createElement('div');
    modalContainer.setAttribute('id', 'modal-portal');
    document.body.appendChild(modalContainer);
  });
  it('should match the snapshot', () => {
    const question = {
      id: 'test-1',
      title: 'Test title',
      subtitle: 'Test subtitle',
      type: singleQuestion.id,
    };

    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          {withDroppable(<QuestionListItem question={question} index={0} />)}
        </IntlProvider>
      </Provider>,
    );

    expect(firstChild).toMatchSnapshot();
  });
});
