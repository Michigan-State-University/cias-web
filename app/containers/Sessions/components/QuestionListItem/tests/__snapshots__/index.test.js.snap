// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionListItem /> should match the snapshot 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 83.33333333333334%;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 8.333333333333332%;
}

.c9 {
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  font-weight: 700;
  color: #2F3850;
}

.c10 {
  margin: 0;
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  color: #6B707C;
}

.c14 {
  width: auto;
  height: auto;
}

.c0 {
  width: auto;
  height: auto;
  display: block;
  border-radius: 5px;
  width: 100%;
}

.c1 {
  width: auto;
  height: auto;
  display: block;
  border-radius: 5px;
  width: 100%;
  padding: 15px;
  margin-bottom: 15px;
  background-color: rgba(248,251,255,1);
  cursor: pointer;
  border-radius: 5px;
  border: 1px solid #EFEFEF;
  border-radius: 5px;
}

.c1:hover {
  background-color: #E9F1FB;
}

.c7 {
  width: auto;
  height: auto;
  display: block;
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c12 {
  width: auto;
  height: auto;
  display: block;
  border-radius: 5px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  justify-items: flex-end;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  justify-items: space-between;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px;
}

.c8 {
  border-radius: 50%;
  background-color: #107969;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 10px;
  height: 10px;
  margin-right: 5px;
  background-color: rgba(16,121,105,1);
  background: #0086F8;
  opacity: 1;
  color: #2F3850;
}

.c5 {
  margin: 0;
  font-family: DM Sans;
  font-size: 15px;
  line-height: 150%;
  margin-bottom: 6px;
  color: #2F3850;
}

.c6 {
  -webkit-line-clamp: 2;
  overflow: hidden;
  -webkit-box-orient: vertical;
  -moz-box-orient: vertical;
  -ms-box-orient: vertical;
  -o-box-orient: vertical;
  display: -webkit-box;
}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:960px) {

}

@media only screen and (min-width:1170px) {

}

@media only screen and (min-width:1360px) {

}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:960px) {

}

@media only screen and (min-width:1170px) {

}

@media only screen and (min-width:1360px) {

}

<div
  data-rbd-droppable-context-id="0"
  data-rbd-droppable-id="TEST"
>
  <div
    aria-describedby="rbd-hidden-text-0-hidden-text-0"
    class="c0"
    data-rbd-drag-handle-context-id="0"
    data-rbd-drag-handle-draggable-id="test-1"
    data-rbd-draggable-context-id="0"
    data-rbd-draggable-id="test-1"
    draggable="false"
    role="button"
    tabindex="0"
    width="100%"
  >
    <div
      class="HoverableBox-sc-zvip6a-0 c1 "
      width="100%"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            class="c4"
          >
            <h3
              class="c5 c6"
            >
              Test subtitle
            </h3>
          </div>
          <div
            class="c4"
          >
            <div
              class="c7"
              display="flex"
            >
              <div
                class="c8"
                opacity="1"
              />
              <p
                class="c9 c10"
                font-weight="bold"
              >
                Single answer
              </p>
            </div>
          </div>
        </div>
        <div
          class="c11"
        >
          <div
            class="c12"
            display="flex"
          >
            <div
              class="c13"
            >
              <img
                alt="dots"
                class="c14"
                src="IMAGE_MOCK"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
